app/v2/controllers/V2MovementsController.scala

```diff
@@ -243,12 +303,38 @@ class V2MovementsControllerImpl @Inject() (
         request.body.runWith(Sink.ignore)
 
         (for {
-          movementResponse <- persistenceService.createMovement(request.eoriNumber, movementType, None).asPresentation
+          movementResponse <- persistenceService.createMovement(request.eoriNumber, movementType, None).asPresentation.leftMap {
+            err =>
+              auditService.auditStatusEvent(
+                CreateMovementDBFailed,
+                Some(Json.toJson(err)),
+                None,
+                None,
+                Some(request.eoriNumber),
+                Some(movementType),
+                None
+              )
+              err
+          }
           upscanResponse <- upscanService
             .upscanInitiate(request.eoriNumber, movementType, movementResponse.movementId, movementResponse.messageId)
             .asPresentation
           boxResponseOption <- mapToOptionalResponse(
-            pushNotificationsService.associate(movementResponse.movementId, movementType, request.headers, request.eoriNumber)
+            pushNotificationsService
+              .associate(movementResponse.movementId, movementType, request.headers, request.eoriNumber)
+              .leftMap {
+                err =>
+                  auditService.auditStatusEvent(
+                    PushNotificationFailed,
+                    Some(Json.obj("message" -> err.toString)),
```

Comment:
here we need to pass ```Json.toJson(err)```

Reply:
This is added on purpose as `err` is not presentation an error - it's a `PushNotificationError`  which is an obj on it's own without any params. To get a proper json we have go with this approach.

Reply:
Ohk my bad. @drnaylor-oc is this ok for you? or need to change it to ```None```?

Reply:
Oh, I see, it's used here. My search skills are bad these days.

Whatever makes the most sense here.